WordPress/Classes/ViewRelated/Blog/SharingDetailViewController.m

```diff
@@ -70,7 +81,7 @@ - (NSManagedObjectContext *)managedObjectContext
 
 - (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
 {
-    if ([self.publicizeConnection isBroken]) {
+    if ([self.publicizeConnection isBroken] || [self mustDisconnectFacebook]) {
```

Comment:
This boolean check is also done in `SharingConnectionsViewController`. In my opinion, that also supports moving this logic to the `PublicizeConnection` object. In this case, also, this boolean check is split in between two classes: part of the logic is defined in the "connection" object, while another part of it is in the `mustDisconnectFacebook` method, which also pulls properties from the connection. What do you think?